core-bundle/contao/widgets/SerpPreview.php

```diff
@@ -46,28 +47,37 @@ public function generate()
 		$title = StringUtil::substr(str_replace(array('&nbsp;', '&shy;'), array(' ', ''), $this->getTitle($model)), 64);
 		$description = StringUtil::substr($this->getDescription($model), 160);
 		$alias = $this->getAlias($model);
+		$url = '';
+		$trail = '';
 
 		try
 		{
-			// Get the URL with a %s placeholder for the alias or ID
+			// Get the URL with an %s placeholder for the alias or ID
 			$url = $this->getUrl($model);
 		}
-		catch (ExceptionInterface $routingException)
+		catch (RouteParametersException)
 		{
 			return '<div class="serp-preview"><p class="tl_info">' . $GLOBALS['TL_LANG']['MSC']['noSerpPreview'] . '</p></div>';
 		}
-
-		list($baseUrl) = explode('%s', $url);
-		$trail = implode(' › ', $this->convertUrlToItems($baseUrl));
-
-		// Use the base URL for the index page
-		if ($model instanceof PageModel && $alias == 'index')
+		catch (ExceptionInterface)
 		{
-			$url = $trail;
+			// Ignore other routing exceptions
 		}
-		else
+
+		if ($url)
```

Comment:
you could actually move this to the `try` condition, then we wouldn't need the `if ($url)`

Reply:
Yes, I thought about that, too. But isn‘t it better if the `try` block only contains the one method call that actually might throw the exception?

Reply:
I don't think it really matters… 🤷

Reply:
Changed in 5b5de6d69953eb2200130cf61293a49d71b9e07e then.